Sources/NIO/ChannelPipeline.swift

```diff
@@ -907,9 +907,37 @@ extension ChannelPipeline {
         case .after(let handler):
             handlers.reverse()
             individualPosition = .after(handler)
+        }
+
+        let promise = self.eventLoop.makePromise(of: Void.self)
+
+        // Add all the handlers.
+        func addAllHandlersAndComplete() {
+            // Function for reducing to propagate errors.
+            func firstErrorOrSuccess(initial: Result<Void, Error>, current: Result<Void, Error>) ->Result<Void, Error> {
+                switch initial {
+                case .success:
+                    return current
+                case .failure:
+                    return initial
+                }
+            }
 
+            let result = handlers.map {
+                _add($0, position: individualPosition)
+            }.reduce(.success(()), firstErrorOrSuccess)
```

Comment:
Let’s avoid map and reduce here and just use a loop, we don’t need the intermediate arrays and if we’re aiming to speed things up anyway it’s another incremental savings.